xc/xc7/fasm2bels/ioi_models.py

```diff
@@ -23,7 +25,60 @@ def get_ioi_site(db, grid, tile, site):
     return iob_site
 
 
+def cleanup_ilogic(top, site):
+    """
+    Cleans-up an ILOGIC site after its routing is known.
+
+    - IDDR/ODDR bels have both S and R inputs but physically it is only one.
+      When connecting one of them to a const the other one has to be tied to
+      the same const too.
+
+    - IDDR as two physical clock inputs CLK and CLKB. Whenever a single clock
+      is used the CLKB input has to be connected to the inverted CLK clock.
+    """
+
+    # Check if we have an IDDR
+    bel = site.maybe_get_bel("IDDR")
+    if bel is not None:
+
+        # Check if the SR is const
+        source = top.find_source_from_sink(site, 'SR')
+
+        # If so then connect both 'S' and 'R' of the BEL to the same const.
+        if source in [0, 1]:
```

Comment:
Should this also have a ` and len(source) == 2`?

Reply:
No, that is not the way it works. Here I check whether the `SR` sink is routed to a constant and if so then I assign it to both to `S` and `R` ports of the `IDDR` cell.

`IDDR` cell has `S` and `R` ports but physically they correspond to a single `SR` input and an appropriate bit controlling whether it is set or reset. I've found in the documentation that when neither reset not set are used, both should be connected to a const (logically).

Reply:
Your right, resolving.